### DISCUSSION TITLE
Updated OpenLayers dependency leads to failing installation

### DISCUSSION THREAD

**Discussion by fhandke-fugro** [biigle/core#954] [General]
Hello,

I noticed when running the install of biigle (build.sh) I get the error:

> > [ 4/17] RUN echo "//npm.pkg.github.com/:_authToken=ghp_dggdg" > .npmrc     && npm install     && MIX_PUSHER_APP_KEY=biigle         npm ru
> n prod     && rm -r .npmrc node_modules:
> 5.647 npm warn deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
> 17.07 npm error code E409
> 17.08 npm error 409 Conflict - GET https://npm.pkg.github.com/download/@biigle/ol/9.2.4/1c025faf01d9f35d5cc036d4c6bc48538a3ba7d1 - Package file checksum mismatch
> 17.08 npm error A complete log of this run can be found in: /root/.npm/_logs/2024-10-17T09_31_03_781Z-debug-0.log
> ------
>  
> 5 warnings found (use --debug to expand):
> - MaintainerDeprecated: Maintainer instruction is deprecated in favor of using label (line 2)
> - SecretsUsedInArgOrEnv: Do not use ARG or ENV instructions for sensitive data (ARG "GITHUB_OAUTH_TOKEN") (line 15)
> - SecretsUsedInArgOrEnv: Do not use ARG or ENV instructions for sensitive data (ARG "MIX_PUSHER_APP_KEY") (line 16)
> - LegacyKeyValueFormat: "ENV key=value" should be used instead of legacy "ENV key value" format (line 36)
> - LegacyKeyValueFormat: "ENV key=value" should be used instead of legacy "ENV key value" format (line 37)
> build.dockerfile:18
> --------------------
>   17 |     # Compile assets. npm is installed above.
>   18 | >>> RUN echo "//npm.pkg.github.com/:_authToken=${GITHUB_OAUTH_TOKEN}" > .npmrc \
>   19 | >>>     && npm install \
>   20 | >>>     && MIX_PUSHER_APP_KEY=${MIX_PUSHER_APP_KEY} \
>   21 | >>>         npm run prod \
>   22 | >>>     && rm -r .npmrc node_modules
>   23 |
> --------------------
> ERROR: failed to solve: process "/bin/sh -c echo \"//npm.pkg.github.com/:_authToken=${GITHUB_OAUTH_TOKEN}\" > .npmrc     && npm install     && MIX_PUSHER_APP_KEY=${MIX_PUSHER_APP_KEY}         npm run prod     && rm -r .npmrc node_modules" did not complete successfully: exit code: 1
> BIIGLE build script executed.

It seems like the biigle/ol package is corruped "https://npm.pkg.github.com/download/@biigle/ol/9.2.4/1c025faf01d9f35d5cc036d4c6bc48538a3ba7d1 - Package file checksum mismatch".

When I checked on the used package in https://github.com/biigle/openlayers/pkgs/npm/ol I noticed that it got updated two days ago from 5.3.4 to 9.2.4. That should be the reason.

Was there a valid reason to update and it should work despite that? Maybe you can spot what I have to change to make it work again.

Or maybe it is something that is the case for everyone. In this case: Could you downgrade to 5.3.4 again for now?

Thanks!

Best,
Felix

**Comment by mzur** [comment]
I fixed this issue a couple of hours ago. The fix was just released. Once the Docker images [were built](https://github.com/biigle/core/actions/runs/11383899200) you can upgrade to v3.86.3.

**Reply by fhandke-fugro** [reply]
Thanks @mzur . That worked. However now that I built the app again I noticed that the API changed a bit.

So would like to go back to a version before that change. Is that easily possible? I build using the ./build.sh command
Thanks!

**Reply by mzur** [reply]
Which API change do you mean? I try to keep most changes backwards-compatible.

Choosing the version is not built into the script, yet. You have to modify all dockerfiles (except websockets) manually like this:

```diff
-FROM ghcr.io/biigle/app:latest
+FROM ghcr.io/biigle/app:v3.86.3
```
(same for `biigle/worker` and `biigle/web`)

**Reply by fhandke-fugro** [reply]
Thanks. I will try that.

One I noticed that does not work anymore is this one: 
![image](https://github.com/user-attachments/assets/c3197242-53b1-451d-9bc3-9be0f2796edc)

**Reply by mzur** [reply]
I was extra careful to keep this API endpoint backwards compatible. Is there any error message when you use it?

**Reply by fhandke-fugro** [reply]
Turns out I did not run ./artisan migrate. Seems to be all good now. Thanks!

**Reply by fhandke-fugro** [reply]
The only thing missing is adding ifdo files. But that was removed on purpose?

**Reply by mzur** [reply]
No this is the part that was changed/enhanced. With iFDOs you have to go the "create new volume (v2)" route. You first create a [pending volume](https://biigle.de/doc/api/index.html#api-Volumes-StoreProjectPendingVolumes) which accepts an iFDO `metadata_file` with the `Biigle\Modules\MetadataIfdo\IfdoParser` `metadata_parser`. In a custom instance, you have to install the [biigle/metadata-ifdo](https://github.com/biigle/metadata-ifdo) module for this. Next, you can finish creating the new volume with [this endpoint](https://biigle.de/doc/api/index.html#api-Volumes-UpdatePendingVolume). The biggest enhancement now is that you can also choose in the second API call to import annotations or file labels from the iFDO file. Also, (only) the new iFDOv2 format is supported.

**Reply by mzur** [reply]
There is also [biigle/metadata-coco](https://github.com/biigle/metadata-coco) as a new alternative to the iFDO annotation import if you do a lot of machine learning.

**Reply by fhandke-fugro** [reply]
Will try that out. Thanks!
